projects/plugins/jetpack/_inc/lib/class-jetpack-top-posts-helper.php

```diff
@@ -137,6 +144,10 @@ function ( $item ) use ( $acceptable_types ) {
 			$top_posts = array_slice( $top_posts, 0, $items_count );
 		}
 
+		if ( $cached ) {
+			set_transient( $transient_key, $top_posts, DAY_IN_SECONDS );
```

Comment:
A day may be a bit too long, folks will expect top posts to be updated after they publish a post, and if that post immediately gets some traffic from social media for example. 

Maybe we could use the same expiration as when fetching the top posts from WordPress.com?
https://github.com/Automattic/jetpack/blob/e9a49af5ddd83d2a0e282178f2b11e9fdfcbb8f5/projects/packages/stats/src/class-wpcom-stats.php#L455-L468

It's currently set to 5 minutes by default but can be overwritten with the filter.

Reply:
I'm fine with adjusting it to whenever -- I'm just hoping to have it ease up on the db load on every page view.

Reply:
```suggestion
			$minutes = class_exists( 'WPCOM_Stats' ) ? WPCOM_Stats::STATS_CACHE_EXPIRATION_IN_MINUTES : 5;
			set_transient( $transient_key, $top_posts, $minutes * MINUTE_IN_SECONDS  );
```

If we wanted to safely rely on the `WPCOM_Stats` class's constant with a fallback.  I'm not sure if there's a situation where it wouldn't be available.

Reply:
My only real concern there is what frequency of churn we'd want to go into the options table, as it would be db writes on the front-end if there isn't a storage for transients in memcached or redis or the like.

Reply:
> If we wanted to safely rely on the WPCOM_Stats class's constant with a fallback. I'm not sure if there's a situation where it wouldn't be available.

What do you think about creating a new public method in `WPCOM_Stats`, that would return `self::STATS_CACHE_EXPIRATION_IN_MINUTES * MINUTE_IN_SECONDS` wrapped in the existing `jetpack_fetch_stats_cache_expiration` filter? You'd then be able to call that from `Jetpack_Top_Posts_Helper` (no need for the `class_exists` check, the class is autoloaded) and you'd be able to customize the expiration from a functionality plugin on your client's side if 5 minutes is still too much.

Reply:
Sure, sounds reasonable.